.github/workflows/publish-release-reusable.yml

```diff
@@ -0,0 +1,89 @@
+name: Publish Draft Release
+
+on:
+  workflow_call:
+    inputs:
+      sha:
+        required: true
+        type: string
+      version:
+        required: true
+        type: string
+      draft:
+        required: true
+        type: boolean
+      release_notes:
+        required: true
+        type: string
+
+env:
+  dotnet-version: 6.0.x # SDK Version for building Dafny
+  
+jobs:
+  check-deep-tests:
+    uses: ./.github/workflows/check-deep-tests-reusable.yml
```

Comment:
I'm happy with the rename of `deep-tests.yml` to `nightly.yml`, but you'll also need to rename and fix the action that explicitly checks for the `deep-tests.yml` workflow file name: https://github.com/dafny-lang/dafny/blob/master/.github/workflows/check-deep-tests-reusable.yml#L25